src/runner.ts

```diff
@@ -183,6 +185,248 @@ export class CoverageParserRunner {
         });
     }
 
+    private mergeCoberturaReports(reportPaths: string[]): types.CoberturaCoverage | undefined {
+        if (!reportPaths.length) {
+            return undefined;
+        }
+
+        let baseReportPath: string = reportPaths[0];
+        core.debug(messagesFormatter.format(messages.using_cobertura_report_as_base_report, baseReportPath));
+
+        let baseCoverage = this.processXMLToObj(baseReportPath);
+        if (reportPaths.length > 1) {
+            for (let i = 1; i < reportPaths.length; i++) {
+                const reportToMerge: types.CoberturaCoverage = this.processXMLToObj(reportPaths[i]);
+                try {
+                    core.debug(messagesFormatter.format(messages.merging_cobertura_report, reportPaths[i]));
+                    baseCoverage = this.mergeCoberturaCoverage(lodash.cloneDeep(baseCoverage), reportToMerge);
+                } catch (error) {
+                    if (error instanceof Error) {
+                        core.warning(messagesFormatter.format(messages.coverage_data_was_not_merged_due_to, reportPaths[i], error.message));
+                    } else {
+                        core.warning(messagesFormatter.format(messages.coverage_data_was_not_merged, reportPaths[i])); // Should never happen
```

Comment:
The error message should include the error object as a string, following the same pattern we used in `main.ts`.